### PR TITLE
Document plugin-publish 0.10.0 features

### PR DESCRIPTION
**Pull request by marcphilipp**
- Integration with `java-gradle-plugin`
- POM customization using `java-gradle-plugin` and `maven-publish`

Resolves gradle/gradle#5987.

### CODE REVIEW

**Comment by marcphilipp** [comment]
@oehme Thanks for the feedback! I'll do the following:

- [ ] Remove the `maven-publish` section from the guide as there's no low-level documentation for the plugin.
- [ ] Move the `java-gradle-plugin` usage to the beginning and present it as the canonical/idiomatic way to write a plugin.
- [ ] I'll explicitly mention 4.10 and will update the guide once it's released.

**Comment by marcphilipp** [comment]
@oehme PTAL! 🙂

File: contents/index.adoc [hunk 1]
```diff
@@ -53,9 +53,10 @@ Add the {publishplugin} to the `plugins` block.
 .build.gradle
 [source,groovy]
 ----
-include::{exampledir}/build.gradle[tags=plugins_block,indent=0]
+include::{exampledir}/build.gradle[tags=plugins_block]
 ----
-<1> The latest version can be found on the https://plugins.gradle.org/plugin/com.gradle.plugin-publish[portal].
+<1> This guide assumes you're using the https://docs.gradle.org/current/userguide/java_gradle_plugin.html[Java Gradle Plugin Development Plugin] which comes with the Gradle distribution and is the recommended way of authoring Gradle plugins.
+<2> The latest version can be found on the https://plugins.gradle.org/plugin/com.gradle.plugin-publish[{portal}].
```
Comment (oehme) [inline contents/index.adoc]: This now reads a bit oddly, as if refering back to point 1

Comment (marcphilipp) [reply contents/index.adoc]: Point taken and improved in 46e2ecc.

**Pull request merged**